pkg/asset/installconfig/gcp/validation.go

```diff
@@ -0,0 +1,94 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	compute "google.golang.org/api/compute/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+type validationData struct {
+	client    API
+	ic        *types.InstallConfig
+	fieldPath *field.Path
+}
+
+// Validate executes platform-specific validation.
+func Validate(client API, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+	vd := &validationData{
+		client:    client,
+		ic:        ic,
+		fieldPath: field.NewPath("platform").Child("gcp"),
+	}
+	allErrs = append(allErrs, validateVPC(vd)...)
+
+	return allErrs
+}
+
+// validateVPC checks that the user-provided VPC is in the project and the provided subnets are valid.
+func validateVPC(vd *validationData) field.ErrorList {
```

Comment:
nit: validationData as input is not readable, i would really like if we start with explicit arguments.

Reply:
I see your point and can change it back to explicit args. The problem is that the signatures of later functions, like validateSubnets gets very long.